pallets/staking/src/lib.rs

```diff
@@ -1543,6 +1559,23 @@ decl_module! {
 
         fn deposit_event() = default;
 
+        fn on_runtime_upgrade() -> Weight {
+            use frame_support::storage::migration::{ StorageIterator, put_storage_value };
+
+            if StorageVersion::get() == Releases::V4_0_0 {
+                StorageIterator::<()>::new(b"Staking", b"PermissionedValidators")
```

Comment:
Let's add a version of `migrate_double_map_keys` for single maps and use it here.

Reply:
You can use `StorageKeyIterator` that decodes the keys itself and doesn't require reversing or decoding inline.